lib/blockscore/base.rb

```diff
@@ -0,0 +1,59 @@
+require 'active_support/core_ext/class'
+require 'active_support/core_ext/string/inflections'
+require 'blockscore/connection'
+
+module BlockScore
+  class Base
+    extend BlockScore::Connection
+
+    class_attribute :api_key, :resource, :version
+
+    def self.inherited(base)
+      base.resource = base.to_s.split('::').last.underscore
+    end
+
+    def initialize(options = {})
+      options[:class] = resource
+      @attrs = options
+    end
+
+    def refresh
+      self.class.retrieve(id)
+      instance_variable_set(:@attrs, r.instance_variable_get(:@attrs))
```

Comment:
I agree. My question is whether or not `attr_reader` is the right fit given my previous comment. What do you think?